tests/test_cluster.go

```diff
@@ -252,6 +258,70 @@ func NewCluster(options *TestClusterConfig, logger log.Logger) (*TestCluster, er
 	return &TestCluster{testBase: testBase, archiverBase: archiverBase, host: cluster}, nil
 }
 
+func setupIndex(esConfig *esclient.Config, logger log.Logger) error {
+	esClient, err := esclient.NewIntegrationTestsClient(esConfig, logger)
+	if err != nil {
+		return err
+	}
+
+	exists, err := esClient.IndexExists(context.Background(), esConfig.GetVisibilityIndex())
+	if err != nil {
+		return err
+	}
+	if exists {
+		logger.Info("Index already exists.", tag.ESIndex(esConfig.GetVisibilityIndex()))
```

Comment:
`logger.Warn` maybe? It seems that it should never happen.
However, I did see sometimes integration tests failing because index already existed.

Reply:
This is what I initially tried to address in this PR. It shouldn't happen in CI anymore. But locally, in debugging, if you stop debugger, for example, tests are not teared down properly.